components/datetime/src/provider/packed_pattern.rs

```diff
@@ -20,6 +20,8 @@ use crate::pattern::runtime::Pattern;
 
 /// A field of [`PackedPatternsBuilder`].
 #[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))] // human-readable only
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))] // human-readable only
```

Comment:
You say this, but there's nothing stopping anyone from using this directly in a data struct, or with a non-human-readable format.

Reply:
It is a type exported in a provider module.

Reply:
That doesn't stop anyone from using it in a data struct

Reply:
I'm closing this PR based on the other comments, but I don't think I agree with the premise "That doesn't stop anyone from using it in a data struct": I think it is our duty to help prevent common bugs in _clients of ICU4X_, but it seems wrong to restrict ICU4X development for the sake of preventing _ICU4X developers_ from doing that. We have a high standard for code with thorough review processes and CI in place.